### ISSUE TITLE
A false negative about the rule NestedForDepth

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#11035]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#NestedForDepth
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run

```bash
/var/tmp $ javac Test.java
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="NestedForDepth"/>
  </module>
</module>

/var/tmp $ cat Test.java
import java.util.*;
public class Test {
    public void foo() {
        ArrayList<Integer> list = new ArrayList<>();
        list.add(1);
        list.add(2);
        list.forEach((i) -> {
            for(int j = 0; j < 2; j++) {
              for(int k = 0; k < 2; k++) { // can this be violation ??
                // should report a warning
              }
            }
        });
        for(Integer i : list) {
          for(int j = 0; j < 2; j++) {
            for(int k = 0; k < 2; k++) { // violation
              // can be detected
            }
          }
        }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:16:13: Nested for depth is 2 (max allowed is 1). [NestedForDepth]
Audit done.
Checkstyle ends with 1 errors.
```

---
**Describe what you expect in detail.**
---
Hi, I found this rule is also caused by new java characteristic. Checkstyle should report a warning at line 9, but it doesn't. However, the code commented below can be detected and they care equivalent. Thanks for your consideration.

**Comment by nmancus1** [comment]
@Zustin please uncomment code in your examples so that we can see exact violation in CLI output. This applies to https://github.com/checkstyle/checkstyle/issues/11033, as well.

**Comment by Zustin** [comment]
@nmancus1 Hi, I have updated them, thanks for your prompt reply and kind consideration.

**Comment by romani** [comment]
from specification of Check:
> Restricts nested for blocks to a specified depth.

forEach method ArrayList class is not a `for` block. There could be numerous methods that do kind of `forEach`. We can not track them all.

You might use https://checkstyle.sourceforge.io/config_misc.html#DescendantToken
to catch that lambda will not have more than 2 `for`.
Or do special Xpath matching by https://checkstyle.sourceforge.io/config_coding.html#MatchXpath to catch what you want.
